contributed_definitions/nyaml/NXopt.yaml

```
@@ -56,348 +43,675 @@ NXopt(NXobject):
+            ENTRY/INSTRUMENT/calibration/calibration_time.
+          enumeration: [calibration time provided, no calibration, within 1 hour, within 1 day, within 1 week]
         calibration_time(NX_DATE_TIME):
           exists: optional
           doc: |
             If calibtration status is 'calibration time provided', specify the
             ISO8601 date when calibration was last performed before this
             measurement. UTC offset should be specified.
-        calibration_data_link:
+        calibration_data_link(NXidentifier):
+          exists: recommended
           doc: |
             Link to the NeXus file containing the calibration data and metadata.
-      (NXbeam_path):
+      # Discussion
+      # ##########
+      # The basic idea was: You have always an incident source and a detector.
+      # Therefore, you always have an incident beam and a "detection beam".
+      # These beams can be assigned clear incident and detection angles.
+      #
+      # This information is always of relevance for ellipsometry,
+      # but for Raman experiments, the state of polarization is as well of relevance
+      # Additonally, this is limited for incident, detection and sample normal
+      # to be in one plane. This does not have to be the case for Raman scattering
+      # experiments. Therefore, this is not general enough and may be moved to
+      # NXellipsometry. 
+      # This information may be as well stored in an incident and detection beam
+
+      # beam properties
+      beam_TYPE(NXbeam):
+        exists: [min, 0, max, infty]
         doc: |
-          Describes an arrangement of optical or other elements, e.g. the beam
-          path between the light source and the sample, or between the sample
-          and the detector unit (including the sources and detectors
-          themselves).
-          
-          If a beam splitter (i.e. a device that splits the incoming beam into
-          two or more beams) is part of the beam path, two or more NXbeam_path
-          fields may be needed to fully describe the beam paths and the correct
-          sequence of the beam path elements.
-          Use as many beam paths as needed to describe the setup.
+          Beam characteristics between two beam_devices.
+
+          It is recommended, to at least define one incident and one detection beam.
+          If this beam is directly connected to an source, chose here the same
+          name appendix as for the NXsource (e.g. TYPE=532nm)
+        incident_wavelength(NX_FLOAT):
+          unit: NX_WAVELENGTH
+        incident_wavelength_spread(NX_NUMBER):
+          exists: recommended
+          unit: NX_WAVELENGTH
+        incident_polarization(NX_NUMBER):
+          exists: recommended
+          unit: NX_ANY
+        extent(NX_FLOAT):
+          exists: recommended
+        associated_source(NX_CHAR):
+          exists: recommended
+          doc: |
+            The path to a the source, which emitted this beam.
```

```suggestion
            The path to the source which emitted this beam.
```
This can be tricky here since you want to describe the beam at many different locations. For most of them, the source is not actually important anymore. So maybe make this optional.